### ISSUE TITLE
vscode "Problems" diagnostic entries lack "source" field

### ISSUE THREAD

**Issue by shoffmeister** [spring-projects/sts4#725]
**Describe the bug**

The vscode extension sometimes emits diagnostic entries into the "Problems" area. These entries are missing the `source` field - and this makes it difficult to determine that the reported problem originates from this extension here.

**To Reproduce**

In a Spring Boot project, create an application.yaml file with this content:

```
---
performance:
  max-internal-processing-concurrency: 10
```

Observe the rendering of the Problems entry created by this extension:

![image](https://user-images.githubusercontent.com/3868036/152114626-95063e38-afb1-4c8c-9f7f-357b43b8c5ef.png)

//exp: information on screen that flags this problem as a Spring Boot problem
//act: nothing suggests that this is a Spring Boot problem

Root cause is simply that, when emitting the diagnostic entry into the vscode host, 

```
[{
	"resource": " ... /src/main/resources/application.yml",
	"owner": "_generated_diagnostic_collection_name_#5",
	"code": "YAML_UNKNOWN_PROPERTY",
	"severity": 4,
	"message": "Unknown property 'performance'",
	"startLineNumber": 2,
	"startColumn": 1,
	"endLineNumber": 2,
	"endColumn": 12
}]
```

this extension fails to emit the `source` field which vscode uses to flag the originator.

Please add unique identifying content into a `source` field

**Sample**

A link to a GitHub repository with a [minimal, reproducible sample](https://stackoverflow.com/help/minimal-reproducible-example).

Reports that include a sample will take priority over reports that do not.
At times, we may require a sample, so it is good to try and include a sample up front.

**Comment by BoykoAlex** [comment]
I'm going to put extension id in the source field for starters.

**Comment by BoykoAlex** [comment]
Fixed with cbdd11831fcc22a1f3d7340d530ba4aaf3ee59a7
